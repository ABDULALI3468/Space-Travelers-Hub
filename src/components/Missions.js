import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Table from 'react-bootstrap/Table';
import { retrieveMissions } from '../Redux/missions/missions';
import Mission from './Mission';
import '../styles/Missions.css';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(retrieveMissions());
    }
  }, []);

  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th className="hide">Hello</th>
        </tr>
      </thead>
      <tbody>
        {
          missions.map((mission) => <Mission key={mission.mission_id} mission={mission} />)
          }
      </tbody>
    </Table>
  );
};

export default Missions;
