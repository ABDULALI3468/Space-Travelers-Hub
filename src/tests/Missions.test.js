import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';

import store from '../Redux/ConfigureStore';

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Missions /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Missions component', () => {
  render(<Provider store={store}><Missions /></Provider>);
  screen.debug();
});
