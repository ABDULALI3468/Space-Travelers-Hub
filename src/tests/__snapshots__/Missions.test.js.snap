// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th
        className="hide"
      >
        Hello
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
